cmd/build/v2/build.go

```diff
@@ -341,13 +343,24 @@ func (bc *OktetoBuilder) buildSvcFromDockerfile(ctx context.Context, manifest *m
 	bc.ioCtrl.Logger().Info(fmt.Sprintf("Building service '%s' from Dockerfile", svcName))
 	isStackManifest := manifest.Type == model.StackType
 	buildSvcInfo := bc.getBuildInfoWithoutVolumeMounts(manifest.Build[svcName], isStackManifest)
-	var err error
 	var buildHash string
 	if bc.smartBuildCtrl.IsEnabled() {
 		buildHash = bc.smartBuildCtrl.GetBuildHash(buildSvcInfo, svcName)
 	}
-	tagToBuild := newImageTagger(bc.Config, bc.smartBuildCtrl).getServiceImageReference(manifest.Name, svcName, buildSvcInfo, buildHash)
-	buildSvcInfo.Image = tagToBuild
+	it := newImageTagger(bc.Config, bc.smartBuildCtrl)
+	tagsToBuild := it.getServiceImageReference(manifest.Name, svcName, buildSvcInfo, buildHash)
+	imageCtrl := registry.NewImageCtrl(bc.oktetoContext)
+	globalImage := it.getGlobalTagFromDevIfNeccesary(tagsToBuild, bc.oktetoContext.GetNamespace(), bc.oktetoContext.GetRegistryURL(), buildHash, manifest.Name, svcName, imageCtrl)
+	if globalImage != "" {
+		tagsToBuild = fmt.Sprintf("%s,%s", tagsToBuild, globalImage)
+	}
+	if buildSvcInfo.Image == "" {
+		devImage := it.getDevTagFromGlobalIfNeccesary(tagsToBuild, bc.oktetoContext.GetNamespace(), bc.oktetoContext.GetGlobalNamespace(), bc.oktetoContext.GetRegistryURL(), buildHash, manifest.Name, svcName, imageCtrl)
+		if devImage != "" {
+			tagsToBuild = fmt.Sprintf("%s,%s", tagsToBuild, devImage)
```

Comment:
Doing this in that order, it will imply that [here](https://github.com/okteto/okteto/pull/4270/files#diff-2bcc58eee26848ca1af49b8495ad2e33224ada8c2be4a3439596d5f3bea42c6fR102) we will display the global image when the image is calculated by us, but we should display the dev one

<img width="1854" alt="Screenshot 2024-05-30 at 09 44 49" src="https://github.com/okteto/okteto/assets/3510171/ced39367-468e-4d1d-a47f-712d62521fa0">

Check in this screenshot which image is being displayed